firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -562,11 +562,32 @@ class BrowserCoordinator: BaseCoordinator,
         )
     }
 
+    func showAddressAutofill(frame: WKFrameInfo?) {
+        let bottomSheetCoordinator = makeAddressAutofillCoordinator()
+        bottomSheetCoordinator.showAddressAutofill(frame: frame)
+    }
+
     func showRequiredPassCode() {
         let bottomSheetCoordinator = makeCredentialAutofillCoordinator()
         bottomSheetCoordinator.showPassCodeController()
     }
 
+    private func makeAddressAutofillCoordinator() -> AddressAutofillCoordinator {
+        if let bottomSheetCoordinator = childCoordinators.first(where: {
+            $0 is AddressAutofillCoordinator
+        }) as? AddressAutofillCoordinator {
+            return bottomSheetCoordinator
+        }
+        let bottomSheetCoordinator = AddressAutofillCoordinator(
```

Comment:
Just to make sure, we need to be careful when we spin off coordinators, they need to be properly `deinit` whenever the VC is removed from the view hierarchy (i.e. when the VC is dismissed, the corresponding coordinator should be dismissed as well). There are some exceptions in our codebase where _some_ coordinators are kept in memory once created, but they are an exception to alleviate some previous technical debt. 

I would expect unit tests for any new coordinator code added in `BrowserCoordinator` and new coordinators like the `AddressAutoFillCoordinator`.

Reply:
Added test